File: ruby/number_of_matches.rb
n = 14

def number_of_matches(n, count = 0)
  if n.even?
    matches = n / 2
    count += matches
    teams = n / 2

  elsif n.odd?
    matches = (n - 1 ) / 2
    count += matches
    teams = ((n - 1) / 2 ) + 1
  end

  teams == 1 ? count : number_of_matches(teams, count)
end

p number_of_matches(n)

File: python/numberOfMatches.py
n = 7

def numberOfMatches(n, count = 0):
  if n % 2 == 0:
    matches = n / 2
    count += matches
    teams = n / 2
  elif n % 2 == 1:
    matches = ( n - 1 ) / 2
    count += matches
    teams = (( n - 1 ) / 2 ) + 1

  return count if teams == 1 else numberOfMatches(teams, count)


print(numberOfMatches(n))